### ISSUE TITLE
<!DOCTYPE HTML  vs <!DOCTYPE html>

### ISSUE THREAD

**Issue by eserte** [ericschernoff/HTMLDoc#4]
On some systems the test suite fails:
```
#   Failed test 'HTML format (2/2)'
#   at t/00_compile.t line 393.
#                   '<!DOCTYPE HTML '
#     doesn't match '(?^:<!DOCTYPE html>)'
# Looks like you failed 1 test of 115.
t/00_compile.t .. 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/115 subtests 
```

On my smokers the failing systems are: centos7, centos8, freebsd9, freebsd10, fedora32, debian+ubuntu until stretch.
It passes on freebsd12, freebsd13, debian+ubuntu since buster+bionic.

**Comment by ericschernoff** [comment]
Thank you for letting me know.  I will work on this and do my best to get it patched by Friday.

**Comment by ericschernoff** [comment]
I went ahead and patched (tweaked, really) the test logic so it would work properly on those systems. I have tested it on CentOS 7, and it worked.  I have released the updated module to CPAN just now.

Please test again when you can and let me know if this is still an issue.

Thank you,
-Eric

**Comment by eserte** [comment]
0.19 still fails on some smokers, but now with a different failure:
```
#   Failed test 'HTMLDoc installed'
#   at t/00_compile.t line 6.
#                   undef
#     doesn't match '/^1\.\d+/'
open3: exec of htmldoc   --portrait  --charset iso-8859-1 --footer .1. --size universal --header .t. --quiet  --format pdf --webpage - failed at /home/cpansand/.cpan/build/2020081120/HTML-HTMLDoc-0.19-0/blib/lib/HTML/HTMLDoc.pm line 1161.
# Looks like your test exited with 2 just after 93.
t/00_compile.t .. 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 23/115 subtests 
```
So there's now some indication that the test requires the htmldoc binary, which indeed isn't installed on all of my smokers.

**Comment by ericschernoff** [comment]
Hello,

It is true that this module can not function without 'htmldoc' installed.  It is just a front-end for that software, and I mention this in the POD, and I will make that bold in a future release.

I have tweaked the test script to bail out of the remaining tests if the htmldoc command is not found, to save time and confusing output.  I just released that change to CPAN.  

I do understand that you may not wish to install htmldoc on all your systems.

Cheers,
-Eric

**Comment by eserte** [comment]
In this case it would be better if the missing requirement is already detected in the configuration phase (i.e. while running Makefile.PL resp. Build.PL) and do something which is recognized as N/A by cpan testers (I think it was exiting with a zero value, but not creating a Makefile or so --- details are probably in the cpan testers docs).

**Comment by ericschernoff** [comment]
Hmmm, so I found this:  http://cpanwiki.grango.org/wiki/CPANAuthorNotes and it recommends editing the Build.PL, "exit 0 unless some_dependency_is_met();"

The catch is that I am using Minilla to package and release this module, and it will overwrite Build.PL.  I wish I could specify external dependencies in a cpanfile.

**Comment by eserte** [comment]
To my knowledge the only possibility to specify non-perl dependencies is my module CPAN::Plugin::Sysdeps: https://metacpan.org/pod/CPAN::Plugin::Sysdeps 
However, it does only work with CPAN.pm's fairly new plugin system; other CPAN installers like cpanminus or cpanp are missing this feature.
HTML::HTMLDoc was already supported except for fedora-like systems yet --- this will be changed in the next release (however, for some reason the newest CentOS release 8 does not have htmldoc in its repositories).

**Comment by ericschernoff** [comment]
OK, I think I have a solution.  I dug into the Minilla POD and found a 'requires_external_bin' option for the minil.toml file.  This added a dependency for 'Devel::CheckBin,' but does seem to solve this issue.

I have made that change and also tweaked my POD to really call out that 'htmldoc' is required, and I have just released 0.21 to CPAN.

Thank you again for testing this module.

**Comment by ericschernoff** [comment]
Hello,

It appears that my latest changes did the trick:

https://cpantesters.org/author/E/ECHERNOF.html

Please let me know if any other issues come up. 

Thanks,
Eric
